src/runtime/hexagon_host.cpp

```diff
@@ -482,16 +481,16 @@ WEAK int halide_hexagon_run(void *user_context,
     if (remote_poll_profiler_state) {
         halide_profiler_lock(s);
         const halide_profiler_instance_state *instance = s->instances;
-        // The instance that called this runtime function should be registered.
-        halide_abort_if_false(user_context, instance);
-        if (instance->next) {
-            halide_profiler_unlock(s);
-            error(user_context) << "Hexagon: multiple simultaneous profiled pipelines is unsupported.";
-            return halide_error_code_cannot_profile_pipeline;
-        }
-        s->get_remote_profiler_state = get_remote_profiler_state;
-        if (remote_profiler_set_current_func) {
-            remote_profiler_set_current_func(instance->current_func);
+        if(instance) {
+            if (instance->next) {
```

Comment:
Isn't that why there is an error thrown inside the if (instance->next) block?

Reply:
I meant that the error should be generated upon attempting to assign to `instance->next`. @abadams Implemented this bit of logic actually.

Reply:
Let me check on this

Reply:
I think this piece of code from abadams in halide_profiler_instance_start should already take care of it.

```
           if (s->get_remote_profiler_state) {
                error(user_context) << "Cannot profile pipeline " << pipeline_name
                                    << " while pipeline " << s->instances->pipeline_stats->name
                                    << " is running, because it is running on a device.";
                return halide_error_code_cannot_profile_pipeline;
            }
```

I think it will make sure that only one instance exists for hexagon. So we can let this code be as is?

Reply:
@mcourteaux Can you confirm if I should do anything to address this comment?

Reply:
Well, I was hoping that @abadams could read what I wrote in the first comment, and judge. I have never worked with HVX.

Reply:
@abadams Any thoughts on this?

Reply:
The code in halide_profiler_instance_start prevents profiling anything if there's currently an on-device profiling instance. The code above prevents profiling an on-device instance if there's currently anything being profiled. These are not quite the same thing. If you have an on-device profiling instance and an off-device one, you hit a different check depending on the order they start in.